src/cpp/r/RSexp.cpp

```diff
@@ -1710,28 +1710,40 @@ bool inherits(SEXP object, const char* S3Class)
    return Rf_inherits(object, S3Class);
 }
 
+// Keep in sync with 'nse.primitives' in SessionCodeTools.R
 std::set<std::string> makeNsePrimitives()
 {
-   std::set<std::string> nsePrimitives;
-   nsePrimitives.insert("quote");
-   nsePrimitives.insert("substitute");
-   nsePrimitives.insert("match.call");
-   nsePrimitives.insert("library");
-   nsePrimitives.insert("require");
-   nsePrimitives.insert("enquote");
-   nsePrimitives.insert("bquote");
-   nsePrimitives.insert("expression");
-   nsePrimitives.insert("evalq");
-   nsePrimitives.insert("subset");
-   nsePrimitives.insert("eval.parent");
-   nsePrimitives.insert("sys.call");
-   nsePrimitives.insert("sys.calls");
-   nsePrimitives.insert("sys.frame");
-   nsePrimitives.insert("sys.frames");
-   nsePrimitives.insert("sys.function");
-   nsePrimitives.insert("sys.parent");
-   nsePrimitives.insert("lazy_dots");
-   return nsePrimitives;
+   return {
+      
+      // base R primitives
+      "bquote",
+      "enquote",
+      "eval.parent",
+      "evalq",
+      "expression",
+      "library",
+      "match.call",
+      "quote",
+      "require",
+      "subset",
+      "substitute",
+      "sys.call",
+      "sys.calls",
+      "sys.frame",
+      "sys.frames",
+      "sys.function",
+      "sys.parent",
+
+      // rlang tidy evaluation
+      "enexpr",
+      "enquo",
+      "enquos",
+      "eval_tidy",
+
+      // dplyr
+      "dplyr_quosures"
+      
+   };
```

Comment:
I think a simple `group_by()` is still going to get flagged

`dplyr:::group_by.data.frame` passes the dots through to `group_by_prepare()` and that is the one that `enquos()` it. That's the main thing I worry about with this kind of approach (but don't have another good option at the moment)

Reply:
Same with `dplyr:::nest_by.data.frame` which just forwards dots to `group_by()`

Reply:
Same with `tidyr:::nest.data.frame` which just forwards dots to `tidyr:::nest.tbl_df` which also just forwards dots to `nest_info()`